mapping_parenting_tech/analysis/prototyping/play_store/get_play_store_apps.py

```diff
@@ -0,0 +1,146 @@
+# ---
+# jupyter:
+#   jupytext:
+#     cell_metadata_filter: -all
+#     comment_magics: true
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.13.3
+#   kernelspec:
+#     display_name: 'Python 3.8.12 64-bit (''mapping_parenting_tech'': conda)'
+#     language: python
+#     name: python3
+# ---
+
+# %% [markdown]
+# # Code to extract app ids from [list of] web pages saved from Google Play store
+
+# %% [markdown]
+# Do imports and set file locations
+
+# %%
+from pathlib import Path
+from google_play_scraper import Sort, app, reviews_all
+from tqdm import tqdm
+import re
+import json
+import sys
+
+INPUT_PATH = Path("~/github/mapping_parenting_tech/inputs/data/")
+OUTPUT_PATH = Path("~/github/mapping_parenting_tech/outputs/data/")
+STORE_PATH = "play_store/"
+PLAY_DATA_FILE = "playstore_innerHTML.html"
+OUTPUT_PLAY_IDS_FILE = "play_store_ids.json"
+OUTPUT_PLAY_DETAILS_FILE = "play_store_details.json"
+OUTPUT_PLAY_REVIEWS_FILE = "play_store_reviews.json"
+
+input_data = INPUT_PATH.expanduser() / STORE_PATH / PLAY_DATA_FILE
+
+if input_data.exists():
+    print("Ok to proceed")
+else:
+    print("Didn't find file")
+
+
+# %% [markdown]
+# ## For Play Store
+# Read HTML and extract list of app ids
+
+# %%
+html_doc = open(input_data, "r")
+html = html_doc.read()
+html_doc.close
+
+link_targets = re.findall(r"(?<=href=\"\/store\/apps\/details\?id=)(.*?)(?=\")", html)
+app_ids = list(dict.fromkeys(link_targets))
+output_ids = {"Parenting apps": app_ids}
+print("Apps ids parsed ok")
+
+# %% [markdown]
+# Save app ids to JSON file
+
+# %%
+output_target = OUTPUT_PATH.expanduser() / OUTPUT_PLAY_IDS_FILE
+with open(output_target, "w") as output_file:
+    json.dump(output_ids, output_file)
+output_file.close
+print("App ids saved to", output_target)
+
+# %% [markdown]
+# Retrieve app details
+
+# %%
+all_app_details = dict()
+remove_apps = list()
+review_counter = 0
+most_reviews = {"id": 0, "count": 0}
+
+for app_id in tqdm(output_ids["Parenting apps"], desc="Retrieving app details"):
+    try:
+        app_details = app(
+            app_id, lang="en", country="gb"  # defaults to 'en'  # defaults to 'us'
+        )
+        all_app_details.update({app_id: app_details})
+        review_counter += app_details["reviews"]
+        if app_details["reviews"] > most_reviews["count"]:
+            most_reviews["id"] = app_id
+            most_reviews["count"] = app_details["reviews"]
+
+    except Exception as e:
```

Comment:
Would be good if you can specify what type of exception do you expect here: https://docs.python.org/3/library/exceptions.html

Reply:
I did look at this, but couldn't make it work when I tried to catch the exceptions that were being thrown, which seem to be custom exceptions (?)

Reply:
Will address in a separate issue